### PR TITLE
feat(api): Add http health check api to ClientTrafficPolicy

### LINKED ISSUE #3258: HealthCheck config for HTTP Downstream
*Description*:

A HealthCheck config is required to setup a health check downstream HTTP filter, aiding the load balancer before Envoy in assessing Envoy’s health. Presently, a HealthCheck exists in the BackendTrafficPolicy, but it serves the upstream.

[optional *Relevant Links*:]

https://www.envoyproxy.io/docs/envoy/v1.29.2/configuration/http/http_filters/health_check_filter

### PR DESCRIPTION
**Pull request by aoledk**
**What this PR does / why we need it**:

When envoy HTTP listener is registered to LB, LB will issue health check requests to check whether the HTTP service is healthy, and we should configure LB to check traffic port instead of `19001` like https://github.com/envoyproxy/gateway/issues/3258#issuecomment-2073889689 mentioned.

So EG should let user to configure health check settings for specific HTTP listener, envoy will not pass through health check requests and respond with a 200 or a 503 depending on the current draining state of the server.

**Which issue(s) this PR fixes**:

Fixes #3258

### CODE REVIEW

File: api/v1alpha1/clienttrafficpolicy_types.go [hunk 1]
```diff
@@ -310,6 +316,19 @@ type HTTP2Settings struct {
...
+type HealthCheckSettings struct {
+	// Headers specifies a set of health check request headers to match on. The health check filter
+	// will check a request’s headers against all the specified headers. To specify the health check
+	// endpoint, set the `:path` header to match on.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	Headers []gwapiv1.HTTPHeaderMatch `json:"headers,omitempty"`
```
Comment (arkodg) [inline api/v1alpha1/clienttrafficpolicy_types.go:329]: prefer we only exposed one field here `Path` and internally mapped the value from here to `:path` within `Headers` of https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/health_check_filter#health-check

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by shawnh2 — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]
LGTM! Thanks!

**Review by cnvergence — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]
🚀

**Pull request merged**